src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3660,6 +3671,11 @@ NetworkOPsImp::addAccountHistoryJob(SubAccountHistoryInfoWeak subInfo)
                             *stTxn, meta->getResultTER(), true, curTxLedger);
                         jvTx[jss::meta] = meta->getJson(JsonOptions::none);
                         jvTx[jss::account_history_tx_index] = txHistoryIndex--;
+
+                        if (i + 1 == num_txns ||
```

Comment:
It's needed for an optimization where attestations are submitted in batch.

Reply:
> It's needed for an optimization where attestations are submitted in batch.

I promise to not continue to beat the dead horse.  But I have to give it one more thump.

1. Attestations are no longer submitted in batch.
2. Even if batch attestation submission is coming, there's no reason this change could not go into a different pull request.

I'm not saying this change is not needed.  I'm just saying it doesn't belong in _this_ pull request.  Put it in a different pull request.  The change is _invisible_ in this pull request because XChain is so big.

No, this is not a hill in intend to die on.  Do what you feel you need to do.

Reply:
I hear you, but I'd still like to keep this.